include/rta/supernode.h

```diff
@@ -0,0 +1,142 @@
+
+    /*!
+     * \brief refresh         - get latest blocks from the daemon
+     * \return                - true on success
+     */
+    void refresh();
+
+    /*!
+     * \brief testnet        - to check if wallet is testnet wallet
+     * \return               - true if testnet
+     */
+    bool testnet() const;
+
+
+    /*!
+     * \brief stakeAmount - returns stake amount
+     * \return            - stake amount in atomic units
+     */
+    uint64_t stakeAmount() const;
+    /*!
+     * \brief walletAddress - returns wallet address as string
+     * \return
+     */
+    std::string walletAddress() const;
+
+    /*!
+     * \brief exportKeyImages - exports key images
+     * \param key_images      - destination vector
+     * \return                - true on success
+     */
+    bool exportKeyImages(std::vector<Supernode::KeyImage> &key_images);
+
+
+    /*!
+     * \brief importKeyImages - imports key images
+     * \param key_images      - source vector
+     * \return                - true on success
+     */
+    bool importKeyImages(const std::vector<KeyImage> &key_images);
+
+    /*!
+     * \brief createFromViewOnlyWallet - creates new Supernode object, creates underlying read-only stake wallet
+     * \param path                     - path to wallet files to be created
+     * \param account_public_address   - public address of read-only wallet
+     * \param viewkey                  - private view key
+     * \param testnet                  - testnet flag
+     * \return                         - pointer to Supernode object on success
+     */
+    static Supernode * createFromViewOnlyWallet(const std::string &path,
+                                       const std::string &address,
+                                       const crypto::secret_key& viewkey = crypto::secret_key(), bool testnet = false);
+
+    static Supernode * load(const std::string &wallet_path, const std::string &wallet_password, const std::string &daemon_address, bool testnet = false,
+                                       const std::string &seed_language = std::string());
+
+    /*!
+     * \brief exportViewkey - exports stake wallet private viewkey
+     * \return private viewkey
+     */
+    crypto::secret_key exportViewkey();
+
+    /*!
+     * \brief signMessage - signs message. internally hashes the message and signs the hash
+     * \param msg         - input message
+     * \param signature   - output signature
+     * \return            - true if successfully signed. false if wallet is watch-only
+     */
+    bool signMessage(const std::string &msg, crypto::signature &signature);
+
+    /*!
+     * \brief verifySignature - verifies signature
+     * \param msg             - message to verify
+     * \param signature       - signer's signature
+     * \return                - true if signature valid
+     */
+    bool verifySignature(const std::string &msg, const string &address, const crypto::signature &signature);
```

Comment:
const method?

Please, unify style: `std::string` and `string`.

Reply:
fixed with 48dcf0b32b60a36f5223ab7e0987a071daf7f0cc, 37c37857e084c1751b7737208b4c9a52c410921e